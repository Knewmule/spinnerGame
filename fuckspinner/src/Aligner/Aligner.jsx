import {styled} from 'styled-components';

export const Bluedot = styled.div`

/*Spinner Style */
  /* Blue dot*/
  left: 18%;
  right:0%;
  top:90%;
  bottom:0%; 
    
    position:absolute;
    height:20%;
    width:100%;
    background-color:blue;
    border-radius:100px;
  
`

export const Align = styled.span`

/*Spinner in the circle gets bigger yellowish*/

  height: 40%;
  width:6%;
  background-color:yellow;
  animation: circleit linear {${addit()}} ;


  position:absolute;
  left:50%;
  right:0%;
  top:30%;
  bottom:0%;
  z-index:1;
  transform:rotate(160deg);


/*Spinner Keyframe Animations*/
@keyframes circleit {
  1% {
  transform: rotate(10deg);
  } 
  25% {
    transform: rotate(230deg);
  }
  100%{
    transform: rotate(450deg);
  } 
}
`

export default function Aligner({addsec}){
  function addit(){
    return addsec;
  }
    return (
        <Align >
            <Bluedot />
        </Align>
    )
}
