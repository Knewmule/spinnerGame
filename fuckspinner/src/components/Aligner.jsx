import styled, {keyframes} from 'styled-components';

export const Bluedot = styled.div`

/*Spinner Style */
  /* Blue dot*/
  left: 18%;
  right:0%;
  top:90%;
  bottom:0%; 
    
    position:absolute;
    height:20%;
    width:100%;
    background-color:blue;
    border-radius:100px;
  
`

export const Align = styled.span`

/*Spinner Keyframe Animations*/
@keyframes circleit {
  1% {
  transform: rotate(10deg);
  } 
  25% {
    transform: rotate(230deg);
  }
  100%{
    transform: rotate(450deg);
  } 
}

/*Spinner in the circle gets bigger yellowish*/

  height: 40%;
  width:6%;
  background-color:yellow;
  // animation: circleit linear {${props => props.ansec}} ;
  animation-name: circleit;
 animation-duration: ${props => props.ansec};
 animation-iteration-count: linear;

  position:absolute;
  left:50%;
  right:0%;
  top:30%;
  bottom:0%;
  z-index:1;
  transform:rotate(160deg);



`

export default function Aligner({spun,ansec}){
// let an = {sec:props.ansec};
    return (
        <Align ansec={ansec} >
            <Bluedot />
        </Align>
    );
};
