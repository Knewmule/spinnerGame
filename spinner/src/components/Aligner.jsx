
import React from 'react';
import style from './Aligner.module.css'

export default function Aligner({d1,d2,d3,d4,sec,blueref}){
  const  circleit = ({d1,d2,d3})=>{
    const one = {
      keyframes:{
        "1%":{
        transform: `rotate(${d1+'deg'})`
      },
      "25%":{
        transform: `rotate(${d2+'deg'})`
      },
      "100%":{
        transform: `rotate(${d3+'deg'})`
      }
      }
    }
    return one;
  }
    const animation ={
      animation:`${circleit({d1,d2,d3})} linear ${sec}s`
    }
    const transform ={
      transform:`rotate(${d4}deg)`
    }
    
  
   
    return (
   
    <span style={animation &&transform} className={style.align}>
      <div ref={blueref} className={style.bluedot}>
      </div>
    </span>
   
        
    );
};
