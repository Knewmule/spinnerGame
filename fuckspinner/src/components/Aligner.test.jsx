import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import {Bluedot} from './Aligner';
import { fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';


describe("Aligner Component", () => {
    it("describes Bluedot style", () =>{
      const bluedot = renderer.create(<Bluedot />).toJSON()
    //   const drawoutCirclerotate = shallow(<Drawout id={circlerotate} />)
    //   expect(drawout).toHaveStyleRule('transform'+drawoutCirclerotate)
      expect(bluedot).toHaveStyleRule('left','18%')
      expect(bluedot).toHaveStyleRule('right','0%')
      expect(bluedot).toHaveStyleRule('top','90%')
      expect(bluedot).toHaveStyleRule('bottom','0%')

      expect(bluedot).toHaveStyleRule('height','20%')
      expect(bluedot).toHaveStyleRule('width','100%')
      expect(bluedot).toHaveStyleRule('position','absolute')
      expect(bluedot).toHaveStyleRule('background-color','blue')
      expect(bluedot).toHaveStyleRule('border-radius','100px')
      expect(bluedot).toMatchSnapshot()
    })

  describe("Aligner Animation", () =>{
    it("animates", () =>{
      // First we can get the teaser from the Jest DOM
const teaser = document.querySelector('#alignfire');
// Fire the click event on the teaser and the necessary animation events on the same element
userEvent.click(teaser);
fireEvent.animationStart(teaser);
fireEvent.animationEnd(teaser);
expect(teaser).not.toBeVisible();
    })
  })


    
})