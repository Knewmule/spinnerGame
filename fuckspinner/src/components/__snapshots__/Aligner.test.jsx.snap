// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aligner Component describes Bluedot style 1`] = `
.c0 {
  left: 18%;
  right: 0%;
  top: 90%;
  bottom: 0%;
  position: absolute;
  height: 20%;
  width: 100%;
  background-color: blue;
  border-radius: 100px;
}

<div
  className="c0"
/>
`;
